pkg/router/controller/controller.go

```diff
@@ -97,7 +114,21 @@ func (c *RouterController) HandleEndpoints() {
 	c.lock.Lock()
 	defer c.lock.Unlock()
 
+	// Change the local sync state within the lock to ensure that all
+	// event handlers have the same view of sync state.
+	c.endpointsReplaceRead = c.EndpointsReplaceRead()
```

Comment:
When we hit the true -> false edge here, is there some clearing of the old data affected by `HandleEndpoints` that I'm just seeing?